### PR TITLE
Ignore min version checks for custom builds

### PR DESCRIPTION
**Pull request by Dezash**
Removes minimum version checks for resources in custom builds.

### CODE REVIEW

**Comment by 4O4** [comment]
It might not be a good idea to completely ignore these checks in custom builds. Such silent differences may be confusing for other developers in the future, especially if it's not documented somewhere. 

If these checks are disrupting the development then I believe it's better to just treat errors as warnings on custom builds and also adjust the messages to make it very clear that they are normally errors on release builds.

**Comment by Dezash** [comment]
The thing is that GetVersionSortable() returns 1.5.5-1.00000.0 for custom builds, which means that any new functions that have version checking will always prevent resources from starting. Release versions do not have these errors unless their version is actually below minimum.

**Comment by 4O4** [comment]
IIRC this revision number could be adjusted in `version.h` unless something has changed in this regard in last year or so.

**Comment by Dezash** [comment]
This is what ccw said: 

> Custom builds don't have a fully valid version number (build number is 0 by default), so certain version checks should probably be skipped/modified

I don't see why would it have valid versions. If a custom build contains certain functions, why would you want to prevent developers from using them in scripts? A custom build should always be considered as the latest version. If for some reason there is a version check for some function, but no actual function, there would still be an error message saying: "attempt to call a nil value".

**Comment by 4O4** [comment]
I'm just saying this whole wall of important compatibility checks shouldn't be __silently__ skipped / cut out in this case :) This may have unpleasant consequences for anyone running production or development servers on custom builds.

**Comment by Dezash** [comment]
I would argue whether they are really important. These checks do not affect the upgrade feature in any way and as I already mentioned, this does not completely silence the errors. Using a non existing function will cause Lua to spew out errors. Deprecated functions will also cause warnings. Regarding your production/development server using a custom build concern, if their build does not contain a function, there would probably be no min version requirement for that function either so these checks wouldn't work anyway.

**Comment by patrikjuvonen** [comment]
I believe if we do this it means we can't test if our `min_mta_version` checks work by changing the version of the custom build?

**Comment by qaisjp** [comment]
I agree with @4O4 that it should not be _silent_. 

My suggestion would be to replace the content of this PR with an update to these lines https://github.com/multitheftauto/mtasa-blue/blob/00889dbbc6b9e3d1b6e5641897293f6cfe203eab/Server/mods/deathmatch/logic/CResource.cpp#L745-L751

to make them look something like

```cpp
if (!GetCompatibilityStatus(strStatus))
{
    m_strFailureReason = SString("Not starting resource %s as %s\n", m_strResourceName.c_str(), strStatus.c_str());
#ifdef MTA_DEBUG
    CLogger::LogPrint("[IGNORED IN CUSTOM BUILD] " + m_strFailureReason);
#else
    CLogger::LogPrint(m_strFailureReason);
    m_eState = EResourceState::Loaded;
    return false;
#endif
}
```


Double-so because of @patrikjuvonen's comment

**Review by qaisjp — DISMISSED** [review]
I agree with @4O4 that it should not be _silent_. 

My suggestion would be to replace the content of this PR with an update to these lines https://github.com/multitheftauto/mtasa-blue/blob/00889dbbc6b9e3d1b6e5641897293f6cfe203eab/Server/mods/deathmatch/logic/CResource.cpp#L745-L751

to make them look something like

```cpp
if (!GetCompatibilityStatus(strStatus))
{
    m_strFailureReason = SString("Not starting resource %s as %s\n", m_strResourceName.c_str(), strStatus.c_str());
#ifdef MTA_DEBUG
    CLogger::LogPrint("[IGNORED IN CUSTOM BUILD] " + m_strFailureReason);
#else
    CLogger::LogPrint(m_strFailureReason);
    m_eState = EResourceState::Loaded;
    return false;
#endif
}
```


Double-so because of @patrikjuvonen's comment

**Review by patrikjuvonen — APPROVED** [review]

**Comment by qaisjp** [comment]
Oh btw I've not tested this,

**Comment by Pirulax** [comment]
Whats the blocker @qaisjp ?

**Comment by qaisjp** [comment]
> Whats the blocker @qaisjp ?

Someone needs to test it

**Comment by patrikjuvonen** [comment]
How about just using the build_overrides.h to define when you want to skip version checks?

File: Server/mods/deathmatch/logic/CResource.cpp [hunk 1]
```diff
@@ -745,9 +745,13 @@ bool CResource::Start(std::list<CResource*>* pDependents, bool bManualStart, con
     if (!GetCompatibilityStatus(strStatus))
     {
         m_strFailureReason = SString("Not starting resource %s as %s\n", m_strResourceName.c_str(), strStatus.c_str());
+#ifdef MTA_DEBUG
```
Comment (patrikjuvonen) [inline Server/mods/deathmatch/logic/CResource.cpp:748]: How about just using the build_overrides.h to define when you want to skip version checks, much like here: https://github.com/multitheftauto/mtasa-blue/blob/a0410c68945b6b3d9225c8c90af2bb6b1cda5eaa/Server/core/CDynamicLibrary.cpp#L148

**Review by patrikjuvonen — CHANGES_REQUESTED** [review]

File: Server/mods/deathmatch/logic/CResource.cpp [hunk 2]
```diff
@@ -745,9 +745,13 @@ bool CResource::Start(std::list<CResource*>* pDependents, bool bManualStart, con
...
+        CLogger::LogPrint("[IGNORED IN CUSTOM BUILD] " + m_strFailureReason);
```
Comment (patrikjuvonen) [inline Server/mods/deathmatch/logic/CResource.cpp:749]: Generates build errors on all debug builds: https://github.com/multitheftauto/mtasa-blue/pull/260/checks?check_run_id=1623745281#step:8:5414

**Review by patrikjuvonen — CHANGES_REQUESTED** [review]

**Comment by patrikjuvonen** [comment]
Closed due to inactivity.

**Pull request closed without merging**